store/store.go

```diff
@@ -233,6 +240,125 @@ func (s *SnapUbuntuStoreRepository) checkStoreResponse(resp *http.Response) {
 	}
 }
 
+/*
+purchase encapsulates the purchase data sent to us from the software center agent.
+
+This object type can be received in response to requests about the user's current
+purchases, and also when making purchase requests.
+
+When making a purchase request, the State "InProgress", together with a RedirectTo
+URL may be received. In-this case, the user must be directed to that webpage.
+Additionally, Partner ID may be recieved as an extended header "X-Partner-Id",
+this should be included in the follow-on requests to the redirect URL.
```

Comment:
This functionality is only going to work with smarter clients like gnome-software, which could use an embedded web frame, and would have to result in an error for the snap cli.